apps/hash-graph/bin/cli/src/subcommand/server.rs

```diff
@@ -343,19 +175,29 @@ pub async fn server(args: ServerArgs) -> Result<(), GraphError> {
             tracing::error!(error = ?report, "Failed to connect to database");
             report
         })?;
+    let store = pool
+        .acquire()
+        .await
+        .change_context(GraphError)
+        .attach_printable("Connection to database failed")?;
 
-    #[cfg(not(feature = "type-fetcher"))]
-    stop_gap_setup(&pool).await?;
+    let pool = if args.offline {
+        restore_builtin_types(SnapshotStore::new(store)).await?;
 
-    #[cfg(feature = "type-fetcher")]
-    let pool = FetchingPool::new(
-        pool,
-        (
-            args.type_fetcher_address.type_fetcher_host,
-            args.type_fetcher_address.type_fetcher_port,
-        ),
-        DomainValidator::new(args.allowed_url_domain.clone()),
-    );
+        FetchingPool::new_offline(pool)
+    } else {
+        // `store` is bound to the lifetime of `pool`, which is moved into `FetchingPool`.
+        // The acquire is done regardless to check for a valid database connection.
+        drop(store);
```

Comment:
I don't quite understand why we drop the store here but we don't do that in the offline branch

Reply:
`restore_built_in_types` consumes the store.

Reply:
Ah it takes ownership, makes sense